modyn/supervisor/internal/pipeline_executor/pipeline_executor.py

```diff
@@ -181,18 +184,9 @@ def _handle_new_data_batch(self, batch: list[tuple[int, int, int]]) -> bool:
             {"batch_size": len(batch), "time": self._sw.stop("trigger_inform"), "num_triggers": num_triggers}
         )
 
-        if num_triggers > 0:
-            logger.info(f"There are {num_triggers} triggers in this batch.")
-            self._handle_triggers_within_batch(batch, triggering_indices)
-            return True
-
-        self._sw.start("selector_inform", overwrite=True)
-        selector_log = self.grpc.inform_selector(self.pipeline_id, batch)
-        self.pipeline_log["supervisor"]["selector_informs"].append(
```

Comment:
we call `inform_selector` in two cases:
1. Here, when within a batch there is no trigger.
2. Inside `_handle_triggers_within_batch`, on the remaining data after processing the last trigger.

The logic in two cases is exactly the same. Also we need to update `remaining_data_range` in two cases with the same logic. So I move this part to inside `_handle_triggers_within_batch`.

Reply:
Thank you for simplifying the logic here.